src/pools/thread/fixed.ts

```diff
@@ -25,6 +29,22 @@ export class FixedThreadPool<
   Data = unknown,
   Response = unknown
 > extends AbstractPool<ThreadWorkerWithMessageChannel, Data, Response> {
+  /**
+   * The promise map.
+   *
+   * - `key`: This is the message ID of each submitted task.
+   * - `value`: An object that contains the worker, the resolve function and the reject function.
+   *
+   * When we receive a message from the worker we get a map entry and resolve/reject the promise based on the message.
+   */
+  protected readonly promiseMap: Map<
+    number,
+    PromiseWorkerResponseWrapper<ThreadWorkerWithMessageChannel, Response>
+  > = new Map<
+    number,
+    PromiseWorkerResponseWrapper<ThreadWorkerWithMessageChannel, Response>
+  >()
```

Comment:
Why is it only added to the fixed pool class? 
It's a global performance optimisation, thus should be added to the abstract class.

Reply:
Yes I was asking that, I Will add It to the cluster pools Aldo ;)

Reply:
If you add it to the abstract class, it's available for all pool/worker type.

Reply:
Yes correct